CONTRIBUTING.md

```diff
@@ -164,21 +473,32 @@ New committers are approved by majority vote of the TSC ([see TSC charter](https
         * Use the imperative mood in the subject line
         * Wrap the body at 72 characters
         * Use the body to explain what and why vs. how
-     
-- Example commit message
-
-```
-Fix OOM caused by foo in bar
-
-Foo was pack ratting ByteBuffers causing OOM.
-
-Cherry-pick of c2bf3b0 (https://github.com/foo/bar/pull/123/commits/c2bf3b025ff456921c4c5ff4bcb814740e823061)
-
-Co-authored-by: Foo Bar <foo@bar.com>
-```
+    * Ensure all code is peer reviewed within your own organization or peers before submitting
+    * Implement and address existing feedback before requesting further review
+    * Make a good faith effort to locate example or referential code before requesting someone else direct you towards it
+    * If you see a lack of documentation, create a separate commit with draft documentation to fill the gap
+        * This documentation can be iterated on same as any code commit, demonstrate in real time that you are learning the code section
+    * Implement or modify relevant tests, otherwise provide clear explanation why test updates were not necessary
+    * Tag your PR with affected code areas as best as you can, it’s okay to tag too many, better to cut down irrelevant tags than miss getting input from relevant subject matter experts
 
 - Example PR message
-  - please refer [here](https://github.com/prestodb/presto/blob/master/pull_request_template.md)
+    - please refer [here](https://github.com/prestodb/presto/blob/master/pull_request_template.md)
+
+### What not to do for Pull Requests
+* Submit before getting peer review in your own organization
+* Request review without addressing or implementing previous feedback
+* Asking reviewers to provide examples or code references instead of finding them on your own
+* Protesting lack of documentation for a code section
+    * Instead, review the related code, then draft initial documentation as a separate commit
+* Submit without test cases or clear justification for lack thereof
+
+## Code Reviews
+#### What to do
+* Provide explicit feedback on what is needed or what would be better
+* Delegate to lower tier support if the code needs too much effort to review
```

Comment:
What is lower tier support?

Reply:
I believe when we wrote this we were thinking about Tier 1,2,3
On a spectrum of community support, technical developer advocate, software engineer, senior software engineer
Each of these can delegate one level down
As an example, if someone submits a very complex commit that eventually will need a senior engineer to review, but the code style or lack of documentation etc. is hindering making it an efficient use of time, then it would get kicked back to DA or a community support person.

Publicly we may not need to mention this as this is more of a concern for senior engineers who belong to organizations that are part of the foundation, and not necessarily a public contributor who wants to volunteer time for code reviews.

Reply:
Sounds good, agreed we can keep it out of the public doc